components/scream/src/physics/shoc/shoc_functions_f90.cpp

```diff
@@ -843,6 +843,42 @@ void pblintd(PblintdData& d)
 
 // end _c impls
 
+// Helper function to copy 1d data to and from Device
+template<typename ViewT>
+void copy_to_device(const std::vector<typename ViewT::value_type const*>& data,
```

Comment:
So, the only big difference between this fcn and what's in ekat (beside the fact that you have a fixed size for all views, which is a minor difference) is that your views are `view<Real*>` rather than `view<Pack<T,N>*>`, right?

Would it make sense to put these utilities in `scream/src/share/util`? Something like `scream_deep_copy.hpp`. I suspect other parts of scream might need this at some point.

Also (but not for this PR, I think), we might have a simpler version:
```
template<typename T>
void deep_copy(const view_1d<T>& dst, typename ViewT::value_type const* src);
template<typename T>
void deep_copy(typename ViewT::value_type const* dst, const view_1d<T>& src);

template<typename T>
void deep_copy(const view_2d<T>& dst, typename ViewT::value_type const* src, const int src_stride);
template<typename T>
void deep_copy(typename ViewT::value_type const* dst, const view_2d<T>& src, const int dst_stride);
```
and have the 'vector' version simply call this. We should also use enalbe_if to provide an overload if T=Pack<S,N>, cause in that case we need the 'physical' length, to avoid reading/writing out of bounds.
The single view version can be maybe a tad easier to visually parse, when syncing 1 or 2 views:
```
view_1d a(ncols);
view_2d b(ncols,nlevs), c(ncols,nlevs);
deep_copy(a, a_data);
deep_copy(b, b_data, nlevs);
deep_copy(c, c_data, nlevs);
```
I feel like creating the vector kind of adds a bit of clunkiness, and doesn't save many lines of code for a handful of views.

Reply:
Yes, I think putting this in `scream/src/share/util` makes sense. I'll move it there.